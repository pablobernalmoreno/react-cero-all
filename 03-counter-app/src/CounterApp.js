import React, { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value }) => {
  const [valor, setValor] = useState(value);

  const handleAdd = () => {
    setValor(valor + 1);
  };

  const handleSubstract = () => {
    setValor(valor - 1);
  };

  const handleReset = () => {
    setValor(value);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{valor}</h2>
      <button id="minus" onClick={handleSubstract}>-1</button>
      <button id="reset" onClick={handleReset}>RESET</button>
      <button id='plus' onClick={handleAdd}>+1</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};

CounterApp.defaultProps = {
  value: 0,
};

export default CounterApp;
