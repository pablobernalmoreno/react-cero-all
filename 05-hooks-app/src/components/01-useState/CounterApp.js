import React, { useState } from "react";
import "./counter.css";

export const CounterApp = () => {
  const [counter, setCounter] = useState({
    counter1: 0,
    counter2: 10,
  });

  const { counter1, counter2 } = counter;
  return (
    <>
      <h1>Counter {counter1}</h1>
      <h1>Counter {counter2}</h1>
      <hr />
      <button
        className="btn btn-primary"
        onClick={() => {
          setCounter({ ...counter, counter1: counter1 + 1 });
        }}
      >
        +1
      </button>
    </>
  );
};
