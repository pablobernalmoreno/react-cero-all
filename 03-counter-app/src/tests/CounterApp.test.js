import React from "react";
import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe("Pruebas en <CounterApp />", () => {
  const value = 0;
  let wrapper = shallow(<CounterApp value={value} />);

  beforeEach(() => {
    wrapper = shallow(<CounterApp value={value} />);
  });

  test("debe mostrar <CounterApp /> correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("debe mostrar el valor por defecto de 0", () => {
    const textoH2 = wrapper.find("h2").text();

    expect(textoH2).toBe(value.toString());
  });

  test("debe incrementar con el boton '+1'", () => {
    const btnplus = wrapper.find("#plus");
    btnplus.simulate("click");
    const textoH2 = wrapper.find("h2").text();

    expect(textoH2).toBe((value + 1).toString());
  });

  test("debe decrementar con el boton '-1'", () => {
    const btnminus = wrapper.find("#minus");
    btnminus.simulate("click");
    const textoH2 = wrapper.find("h2").text();

    expect(textoH2).toBe((value - 1).toString());
  });

  test("debe resetear con el boton '0'", () => {
    const btnminus = wrapper.find("#minus");
    btnminus.simulate("click");
    const btnreset = wrapper.find("#reset");
    btnreset.simulate("click");
    const textoH2 = wrapper.find("h2").text();

    expect(textoH2).toBe(value.toString());
  });
});
