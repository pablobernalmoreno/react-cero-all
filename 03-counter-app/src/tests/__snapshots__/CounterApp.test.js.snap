// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    0
  </h2>
  <button
    id="minus"
    onClick={[Function]}
  >
    -1
  </button>
  <button
    id="reset"
    onClick={[Function]}
  >
    RESET
  </button>
  <button
    id="plus"
    onClick={[Function]}
  >
    +1
  </button>
</Fragment>
`;
